framework/include/physics/PhysicsBase.h

```diff
@@ -249,3 +463,177 @@ PhysicsBase::checkVectorParamsNoOverlap(const std::vector<std::string> & param_v
+PhysicsBase::parameterConsistent(const InputParameters & other_param,
+                                 const std::string & param_name,
+                                 bool warn) const
+{
+  assertParamDefined<T>(param_name);
+  mooseAssert(other_param.have_parameter<T>(param_name),
+              "This should have been a parameter from the parameters being compared");
+  bool consistent = true;
+  if (parameters().isParamValid(param_name) && other_param.isParamValid(param_name))
+  {
+    if constexpr (std::is_same_v<MooseEnum, T>)
+    {
+      if (!getParam<T>(param_name).compareCurrent(other_param.get<T>(param_name)))
+        consistent = false;
+    }
+    else if (getParam<T>(param_name) != other_param.get<T>(param_name))
+      consistent = false;
+  }
+  if (warn && !consistent)
+    mooseWarning("Parameter " + param_name + " is inconsistent between Physics \"" + name() +
+                 "\" of type \"" + type() + "\" and the parameter set for \"" +
+                 other_param.get<std::string>("_action_name") + "\" of type \"" +
+                 other_param.get<std::string>("action_type") + "\"");
+  return consistent;
+}
+
+template <typename T>
+void
+PhysicsBase::warnInconsistent(const InputParameters & other_param,
+                              const std::string & param_name) const
+{
+  parameterConsistent<T>(other_param, param_name, true);
+}
+
+template <typename T, typename C>
+std::map<T, C>
+PhysicsBase::createMapFromVectors(std::vector<T> keys, std::vector<C> values) const
+{
+  std::map<T, C> map;
+  // No values have been specified.
+  if (!values.size())
+  {
+    return map;
+    // If we cant return a map of default C, dont try it
+    // if constexpr (std::is_same_v<MooseEnum, T> || std::is_same_v<MultiMooseEnum, T>)
+    //   return map;
+
+    // C def;
+    // for (const auto & k : keys)
+    //   map[k] = def;
+    // return map;
+  }
+  std::transform(keys.begin(),
+                 keys.end(),
+                 values.begin(),
+                 std::inserter(map, map.end()),
+                 [](T a, C b) { return std::make_pair(a, b); });
+  return map;
+}
+
+template <typename T>
+std::map<T, MooseEnum>
+PhysicsBase::createMapFromVectorAndMultiMooseEnum(std::vector<T> keys, MultiMooseEnum values) const
+{
+  std::map<T, MooseEnum> map;
+  // No values have been specified. We cant form a map of empty MooseEnum
+  if (!values.size())
+    return map;
+  std::transform(keys.begin(),
+                 keys.end(),
+                 values.begin(),
+                 std::inserter(map, map.end()),
+                 [values](T a, MooseEnumItem b)
+                 {
+                   // Create a MooseEnum from the available values in the MultiMooseEnum and an
+                   // actual current active item from that same MultiMooseEnum
+                   MooseEnum single_value(values.getRawNames(), b.name());
+                   return std::make_pair(a, single_value);
+                 });
+  return map;
+}
```

Comment:
I think these should go to moose one of the utility files.

Reply:
We should be able to pass in the function arguments as const references and not do all these container copies

Reply:
fixed